### PR TITLE
strictify app/invocation: everything else

### PR DESCRIPTION
**Pull request by jdhollen**
**Related issues**: N/A

### CODE REVIEW

**Comment by jdhollen** [comment]
the downside is the change is big, the upside is this is the last one

File: app/invocation/invocation_artifacts_card.tsx [hunk 1]
```diff
@@ -19,14 +19,14 @@ export default class ArtifactsCardComponent extends React.Component<Props, State
...
-  handleArtifactClicked(outputUri: string, outputFilename: string, event: MouseEvent) {
+  handleArtifactClicked(outputUri: string, outputFilename: string, event: React.MouseEvent<HTMLAnchorElement>) {
     event.preventDefault();
     if (!outputUri) return false;
 
     if (outputUri.startsWith("file://")) {
       window.prompt("Copy artifact path to clipboard: Cmd+C, Enter", outputUri);
     } else if (outputUri.startsWith("bytestream://")) {
-      rpcService.downloadBytestreamFile(outputFilename, outputUri, this.props.model.getId());
+      rpcService.downloadBytestreamFile(outputFilename, outputUri, this.props.model.getId() ?? "");
```
Comment (iain-macdonald) [inline app/invocation/invocation_artifacts_card.tsx:29]: should probably skip this if getId() is undefined no?

File: rules/typescript/index.bzl [hunk 2]
```diff
@@ -10,12 +10,9 @@ def _swc(**kwargs):
     )
 
 def ts_library(name, srcs, strict = True, **kwargs):
```
Comment (iain-macdonald) [inline rules/typescript/index.bzl:12]: mind removing the parameter too?

**Review by bduffany — APPROVED** [review]

File: rules/typescript/index.bzl [hunk 3]
```diff
@@ -10,12 +10,9 @@ def _swc(**kwargs):
...
-    tsconfig = "//:tsconfig"
-    if strict:
-        tsconfig = "//:tsconfig_strict"
     ts_project(
         name = name,
-        tsconfig = tsconfig,
+        tsconfig = "//:tsconfig_strict",
```
[the remark this answers is not in the document]
Comment (jdhollen) [reply rules/typescript/index.bzl]: I reverted this, I'll send a separate change that does the cleanup on the rule.

File: app/invocation/invocation_artifacts_card.tsx (hunk 1 above)
Comment (jdhollen) [reply app/invocation/invocation_artifacts_card.tsx:29]: So, yeah.  We have a lot of these now.  The production behavior of this today builds a URL containing "undefined" (i.e., this path is already unprotected).  Making interactions with invocation_model a bit more rigorous should probably be at the top of the list, but this doesn't make things worse and let's us get the bits flipped, so let's keep it separate for now.

**Pull request merged**